test-infra/scripts/library.sh

```diff
@@ -0,0 +1,670 @@
+  local geoflag="--region=$3"
+  [[ -n $4 ]] && geoflag="--zone=$3-$4"
+  # Get the password of the admin and use it, as the service account (or the user)
+  # might not have the necessary permission.
+  local password=$(gcloud --format="value(masterAuth.password)" \
+      container clusters describe $2 ${geoflag})
+  if [[ -n "${password}" ]]; then
+    # Cluster created with basic authentication
+    kubectl config set-credentials cluster-admin \
+        --username=admin --password=${password}
+  else
+    local cert=$(mktemp)
+    local key=$(mktemp)
+    echo "Certificate in ${cert}, key in ${key}"
+    gcloud --format="value(masterAuth.clientCertificate)" \
+      container clusters describe $2 ${geoflag} | base64 --decode > ${cert}
+    gcloud --format="value(masterAuth.clientKey)" \
+      container clusters describe $2 ${geoflag} | base64 --decode > ${key}
+    kubectl config set-credentials cluster-admin \
+      --client-certificate=${cert} --client-key=${key}
+  fi
+  kubectl config set-context $(kubectl config current-context) \
+      --user=cluster-admin
+  kubectl create clusterrolebinding cluster-admin-binding \
+      --clusterrole=cluster-admin \
+      --user=$1
+  # Reset back to the default account
+  gcloud container clusters get-credentials \
+      $2 ${geoflag} --project $(gcloud config get-value project)
+}
+
+# Run a command through tee and capture its output.
+# Parameters: $1 - file where the output will be stored.
+#             $2... - command to run.
+function capture_output() {
+  local report="$1"
+  shift
+  "$@" 2>&1 | tee "${report}"
+  local failed=( ${PIPESTATUS[@]} )
+  [[ ${failed[0]} -eq 0 ]] && failed=${failed[1]} || failed=${failed[0]}
+  return ${failed}
+}
+
+# Print failed step, which could be highlighted by spyglass.
+# Parameters: $1...n - description of step that failed
+function step_failed() {
+  local spyglass_token="Step failed:"
+  echo "${spyglass_token} $@"
+}
+
+# Create a temporary file with the given extension in a way that works on both Linux and macOS.
+# Parameters: $1 - file name without extension (e.g. 'myfile_XXXX')
+#             $2 - file extension (e.g. 'xml')
+function mktemp_with_extension() {
+  local nameprefix
+  local fullname
+
+  nameprefix="$(mktemp $1)"
+  fullname="${nameprefix}.$2"
+  mv ${nameprefix} ${fullname}
+
+  echo ${fullname}
+}
+
+# Create a JUnit XML for a test.
+# Parameters: $1 - check class name as an identifier (e.g. BuildTests)
+#             $2 - check name as an identifier (e.g., GoBuild)
+#             $3 - failure message (can contain newlines), optional (means success)
+function create_junit_xml() {
+  local xml="$(mktemp_with_extension ${ARTIFACTS}/junit_XXXXXXXX xml)"
+  local failure=""
+  if [[ "$3" != "" ]]; then
+    # Transform newlines into HTML code.
+    # Also escape `<` and `>` as here: https://github.com/golang/go/blob/50bd1c4d4eb4fac8ddeb5f063c099daccfb71b26/src/encoding/json/encode.go#L48, 
```

Comment:
<!--whitespace-bot-->
Remove trailing whitespace:
```suggestion
    # Also escape `<` and `>` as here: https://github.com/golang/go/blob/50bd1c4d4eb4fac8ddeb5f063c099daccfb71b26/src/encoding/json/encode.go#L48,
```

Reply:
@mattmoor is there a way to exclude directories from the bot check ? These files are dependent files which are downloaded and committed here.

Alternatively I could move that also to a "vendor" dir (but doesn't make much sense except for keeping the bot happy).